project/profiles/capacity4more/modules/c4m/user/c4m_user_subscribe/c4m_user_subscribe.install

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Install and update hooks for the c4m_user_subscribe functionality.
+ */
+
+/**
+ * Implements hook_update_N().
+ */
+function c4m_user_subscribe_update_7001(&$sandbox) {
+  $ret = [];
+
+  if (!isset($sandbox['progress'])) {
+    $sandbox['progress'] = 0;
+    $sandbox['max'] = db_query("SELECT COUNT(*) FROM {field_data_c4m_vocab_topic} WHERE bundle = 'user'")->fetchField();
+    $sandbox['messages'] = [];
+  }
+
+  $limit = 100;
+
+  $results = db_select('field_data_c4m_vocab_topic', 'v')
+    ->fields('v', ['entity_id', 'c4m_vocab_topic_tid'])
+    ->condition('v.bundle', 'user')
+    ->range($sandbox['progress'], $limit)
+    ->orderBy('entity_id')
+    ->execute()
+    ->fetchAll();
+
+  $assoc = _c4m_user_subscribe_group_term_ids_by_uid($results, 'entity_id', 'c4m_vocab_topic_tid');
+
+  foreach ($assoc as $uid => $terms) {
+    $account = user_load($uid);
+    _c4m_user_subscribe_user_flag_terms($terms, $account);
+
+    $sandbox['progress'] += count($terms);
+  }
+
+  $sandbox['#finished'] = ($sandbox['progress'] >= $sandbox['max']) ? TRUE : ($sandbox['progress'] / $sandbox['max']);
+
+  if ($sandbox['#finished']) {
+    return t('Completed the update hook and subscribed: @count_users to @count_terms topics of interest subscriptions.',
+      [
+        '@count_users' => $sandbox['progress'],
+        '@count_terms' => $sandbox['max'],
+      ]);
+  }
+}
+
+/**
+ * Implements hook_update_N().
+ */
+function c4m_user_subscribe_update_7002(&$sandbox) {
+  $ret = [];
```

Comment:
The content of this function is very similar to the one above. Only the vocab machine name and the name used in the message differs. Although this code is ran once and doesn't need maintenance, I would create a single function with params which is called twice.